diff --git a/configure b/configure
index f6f5c29..f615947 100755
--- a/configure
+++ b/configure
@@ -6516,14 +6516,18 @@ case "$custom_allocator" in
             malloc_prefix=$($pkg_config --variable=jemalloc_prefix $pkg_config_flags jemalloc 2>/dev/null)
         require_pkg_config custom_allocator jemalloc jemalloc/jemalloc.h ${malloc_prefix}malloc
     ;;
     tcmalloc)
         require_pkg_config custom_allocator libtcmalloc gperftools/tcmalloc.h tc_malloc
         malloc_prefix=tc_
     ;;
+    mimalloc)
+        require_pkg_config custom_allocator mimalloc mimalloc.h mi_malloc
+        malloc_prefix=mi_
+    ;;
     *)
         require_pkg_config custom_allocator "$custom_allocator" stdlib.h malloc
     ;;
 esac
 
 if test -n "$custom_allocator"; then
     add_extralibs "$custom_allocator_extralibs"
